### PR TITLE
RPRBLND-2241: Update Hybrid Pro to support FSR2

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Update Hybrid Pro to support FSR2

### EFFECT OF CHANGE
Now, Vievport Upscaling uses a built-in Hybrid Pro FSR2.
Added Viewport Upscale Quality to choose a quality mode.
image

Viewport Upscaling is available only if Viewport Denoising is enabled.
image

### TECHNICAL STEPS
Added enum property viewport_upscale_quality and UI element for upscaler quality modes. RPR_FSR2_QUALITY_ULTRA_QUALITY
RPR_FSR2_QUALITY_MODE_QUALITY
RPR_FSR2_QUALITY_MODE_BALANCE
RPR_FSR2_QUALITY_MODE_PERFORMANCE
RPR_FSR2_QUALITY_MODE_ULTRA_PERFORMANCE

Overrided setup_upscale_filter, pass additional argument scene to track viewport_upscale_quality updates. Adjusted notify_status.

### CODE REVIEW

**Review by ARozghon — APPROVED** [review]
Approved by QA, all work as expected

**Comment by bsavery** [comment]
@bnagirniak please review

File: src/rprblender/ui/render.py [hunk 1]
```diff
@@ -174,7 +174,14 @@ def draw(self, context):
         col.prop(settings, 'use_gl_interop')
 
         col.prop(rpr, 'viewport_denoiser')
-        col.prop(rpr, 'viewport_upscale')
+        col1 = col.column()
+
+        col1.prop(rpr, 'viewport_upscale')
```
Comment (bnagirniak) [inline src/rprblender/ui/render.py:179]: Disable viewport_upscale_quality if viewport_denoiser is unset.

File: src/rprblender/engine/viewport_engine_hybridpro.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 from . import context_hybridpro
 
 import pyrpr
+import pyrprwrap
```
Comment (bnagirniak) [inline src/rprblender/engine/viewport_engine_hybridpro.py:19]: no need to import pyrprwrap

File: src/rprblender/engine/viewport_engine_hybridpro.py [hunk 3]
```diff
@@ -62,13 +63,26 @@ def setup_image_filter(self, settings):
...
+    def setup_upscale_filter(self, settings):
+        res = False
+        scene = settings['scene']
+
+        # UPSCALER_FSR2 works only if denoiser is enabled
+        self.is_upscaled = settings['enable'] and self.is_denoised
+        if self.is_upscaled:
+            res |= self.rpr_context.set_parameter(pyrpr.CONTEXT_UPSCALER, pyrpr.UPSCALER_FSR2)
+            res |= self.rpr_context.set_parameter(pyrpr.CONTEXT_FSR2_QUALITY,
+                                                 getattr(pyrprwrap, scene.rpr.viewport_upscale_quality))
```
Comment (bnagirniak) [inline src/rprblender/engine/viewport_engine_hybridpro.py:75]: use `pyrpr`

File: src/rprblender/engine/viewport_engine_hybridpro.py [hunk 4]
```diff
@@ -62,13 +63,26 @@ def setup_image_filter(self, settings):
         self.rpr_context.set_parameter(pyrpr.CONTEXT_PT_DENOISER,
                                        pyrpr.DENOISER_SVGF if self.is_denoised else pyrpr.DENOISER_NONE)
 
+    def setup_upscale_filter(self, settings):
+        res = False
+        scene = settings['scene']
```
Comment (bnagirniak) [inline src/rprblender/engine/viewport_engine_hybridpro.py:68]: use `scene = bpy.context.scene`

File: src/rprblender/engine/viewport_engine.py [hunk 5]
```diff
@@ -515,6 +515,7 @@ def sync(self, context, depsgraph):
         self.setup_upscale_filter({
             'enable': scene.rpr.viewport_upscale,
             'resolution': (self.width, self.height),
+            'scene': scene,
```
Comment (bnagirniak) [inline src/rprblender/engine/viewport_engine.py:518]: no need this

**Review by bnagirniak — CHANGES_REQUESTED** [review]

File: src/rprblender/engine/viewport_engine.py (hunk 5 above)
Comment (VascoPi) [reply src/rprblender/engine/viewport_engine.py]: fixed https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/577/commits/bdb00ef37826dc9b26876222f8cd5e316a372e6b

File: src/rprblender/engine/viewport_engine_hybridpro.py (hunk 3 above)
Comment (VascoPi) [reply src/rprblender/engine/viewport_engine_hybridpro.py]: fixed https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/577/commits/bdb00ef37826dc9b26876222f8cd5e316a372e6b

File: src/rprblender/engine/viewport_engine_hybridpro.py (hunk 2 above)
Comment (VascoPi) [reply src/rprblender/engine/viewport_engine_hybridpro.py]: fixed https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/577/commits/bdb00ef37826dc9b26876222f8cd5e316a372e6b

File: src/rprblender/ui/render.py (hunk 1 above)
Comment (VascoPi) [reply src/rprblender/ui/render.py:179]: fixed https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/577/commits/bdb00ef37826dc9b26876222f8cd5e316a372e6b

File: src/rprblender/ui/render.py [hunk 6]
```diff
@@ -174,7 +174,14 @@ def draw(self, context):
...
+        if rpr.render_quality in ('HYBRIDPRO'):
```
Comment (bnagirniak) [inline src/rprblender/ui/render.py:180]: if rpr.render_quality == 'HYBRIDPRO':

Comment (VascoPi) [reply src/rprblender/ui/render.py]: Fixed https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/577/commits/807c7d9475493c8f10e53a7a2759913ab3638d76

**Review by bnagirniak — APPROVED** [review]
Tested - works good

**Review by DagerD — APPROVED** [review]

**Pull request merged**